### PR TITLE
Purchase order barcode fixes

### PR DESCRIPTION
**Pull request by SchrodingersGat**
Follow up to https://github.com/inventree/InvenTree/pull/5509

- Fixes implementation for digikey
- Improve user feedback
- Simplify / refactor implementation

### CODE REVIEW

File: InvenTree/plugin/base/barcodes/mixins.py [hunk 1]
```diff
@@ -258,129 +428,3 @@ def receive_purchase_order_item(
...
-    "1K":  "purchase_order_number",     # noqa: E241  DigiKey uses 1K instead of K
-    "11K": "packing_list_number",       # noqa: E241
-    "6D":  "ship_date",                 # noqa: E241
-    "P":   "supplier_part_number",      # noqa: E241  "Customer Part Number"
-    "1P":  "manufacturer_part_number",  # noqa: E241  "Supplier Part Number"
-    "4K":  "purchase_order_line",       # noqa: E241
-    "14K": "purchase_order_line",       # noqa: E241  Mouser uses 14K instead of 4K
-    "Q":   "quantity",                  # noqa: E241
-    "9D":  "date_yyww",                 # noqa: E241
-    "10D": "date_yyww",                 # noqa: E241
```
Comment (matmair) [inline InvenTree/plugin/base/barcodes/mixins.py:382]: Question: How do you access this data now?

File: InvenTree/plugin/base/barcodes/mixins.py [hunk 2]
```diff
@@ -258,129 +428,3 @@ def receive_purchase_order_item(
...
-        return suppliers.first()
-
-
-# Map ECIA Data Identifier to human readable identifier
-# The following identifiers haven't been implemented: 3S, 4S, 5S, S
-ECIA_DATA_IDENTIFIER_MAP = {
-    "K":   "purchase_order_number",     # noqa: E241
-    "1K":  "purchase_order_number",     # noqa: E241  DigiKey uses 1K instead of K
-    "11K": "packing_list_number",       # noqa: E241
-    "6D":  "ship_date",                 # noqa: E241
```
Comment (matmair) [inline InvenTree/plugin/base/barcodes/mixins.py:375]: Question: How do you access this data now?

File: InvenTree/plugin/base/barcodes/mixins.py [hunk 3]
```diff
@@ -54,60 +53,218 @@ def scan(self, barcode_data):
...
+        # Return the first "match" for the given code
+        for field in fields:
+            if field.startswith(code):
+                return field[len(code):]
+
         return None
 
-    @staticmethod
-    def parse_ecia_barcode2d(barcode_data: str | list[str]) -> dict[str, str]:
-        """Parse a standard ECIA 2D barcode, according to https://www.ecianow.org/assets/docs/ECIA_Specifications.pdf"""
```
Comment (matmair) [inline InvenTree/plugin/base/barcodes/mixins.py:78]: Comment: This function is IMO very helpful, I would leave it in. The function returned a lot more info than the rewrite and is very usefull to extract more info from barcodes.

**Comment by SchrodingersGat** [comment]
Closing. Will wait until https://github.com/inventree/InvenTree/pull/5919 is merged and then reassess

**Pull request closed without merging**